testing/citest/tests/kube_smoke_test.py

```diff
@@ -699,56 +722,58 @@ def execute_delete_statefulset_pipeline(self):
       builder = kube.KubeContractBuilder(self.kube_observer)
       (builder.new_clause_builder('StatefulSet Deleted', retryable_for_secs=15)
        .get_resources(
-          'statefulsets', extra_args=[self.TEST_APP+'tstv001', '--namespace', self.TEST_NAMESPACE])
-       .contains_path_value('kind', 'StatefulSet'))
+          'statefulsets', extra_args=[self.TEST_APP +
+                                      self.POSTFIX_STATEFUL_DAEMONSET, '--namespace',
+                                      self.TEST_NAMESPACE], no_resource_ok=True)
```

Comment:
I'm trying to move away from no_resource_ok=True and be more explicit about expecting an error. It might be that the Kube Agent doesnt support that yet. However here's an example from the google_smoke_test

    (builder.new_clause_builder('Managed Instance Group Removed')
     .inspect_resource('instanceGroupManagers', group_name)
     .EXPECT(ov_factory.error_list_contains(
         gcp.HttpErrorPredicate(http_code=404)))
     .OR(ov_factory.value_list_path_contains('targetSize', jp.NUM_EQ(0))))
where:
import citest.json_contract as jc
ov_factory = jc.ObservationPredicateFactory()

This is more explicit about acceptable outcomes are either a 404 or a size of 0.

The kubectl agent doesnt have any kind of special error handling so you might not be able to do it that way here. Could put something on your backlog to add support to kubectl so you can change this condition to expect particular errors (or targetSize of 0) rather than using no_resource_ok ?

This is ok as is for now because that's all you have to work with and I dont want to hold the test up.